generator/src/main/java/io/github/guegse/foreachstream/generator/TakeWhile.java

```diff
@@ -0,0 +1,32 @@
+package io.github.guegse.foreachstream.generator;
+
+import java.util.List;
+
+public class TakeWhile extends StatefulIntermediateOperation{
+    @Override
+    String getName() {
+        return "takeWhile";
+    }
+
+    @Override
+    String getTargetType(String inputType, String nextTargetType) {
+        return inputType;
+    }
+
+    @Override
+    List<String> getArgumentTypes(String inputType, String nextOutputType) {
+        return List.of(predicateType(inputType));
+    }
+
+    @Override
+    void emitOperation(Emitter out, String inputType, List<String> arguments, String currentStreamElement, String nextTargetType, String nextTargetElement) {
+        out.printIndentation();
+        out.println("if(!" + arguments.get(0) + ".test(" + currentStreamElement + ")) {");
```

Comment:
I think this has the same problem as limit: takeWhile is a short-circuiting operation, therefore it must be evaluated before the other operations

Reply:
In contrast to limit, takeWhile is not evaluated before the other operations, example:
```java
        List.of(1, 2, 3).stream().filter(element -> {
            count.incrementAndGet();
            return true;
        }).takeWhile(element -> false).forEach(System.out::println);
        System.out.println(count.get());
``` 
Output:
```console
1
```